projects/plugins/boost/app/modules/optimizations/cloud-css/Cloud_CSS.php

```diff
@@ -149,7 +155,7 @@ public function regenerate_cloud_css() {
 	 * Called when stored Critical CSS has been invalidated. Triggers a new Cloud CSS request.
 	 */
 	public function handle_critical_css_invalidated() {
-		$this->regenerate_cloud_css();
+		$this->regenerate_cloud_css( self::REGENERATE_REASON_INVALIDATED );
```

Comment:
Looks like this can be called from `Critical_CSS_Invalidator` when the environment changes OR on deactivation. Is it worth tracking which caused it?

Reply:
I moved the hook firing to avoid requesting C.CSS on plugin deactivation. https://github.com/Automattic/jetpack/pull/36519/commits/36a5669811c8cf67924bac0fc5d06c7bc5a1e2e3